### PR TITLE
literal-block-distance based learnt clause removal and learning rate based branching heuristics

### PR DESCRIPTION
**Pull request by anwu1219**
1. Add options: --decision==lrb (learning rate based branching heuristics) and justification-stoponly-lrb
2. Add option: --lbd (literal block distance aggressive conflict clause removal), which is compatible with every branching heuristics (by default, on for lrb and lrb-stoponly, off for the others).

### CODE REVIEW

File: src/prop/minisat/core/Solver.h [hunk 1]
```diff
@@ -252,8 +252,14 @@ class Solver {
     // Mode of operation:
     //
     int       verbosity;
+    // for LRB
+    double step_size;
+    double step_size_dec;
+    double min_step_size;
```
Comment (4tXJ7f) [inline src/prop/minisat/core/Solver.h]: It looks like these three variables are not actually initialized. Am I missing something?

File: src/options/options_handler.cpp [hunk 2]
```diff
@@ -1410,16 +1410,27 @@ justification-stoponly\n\
...
+    options::decisionStopOnly.set(2);
+    return decision::DECISION_STRATEGY_JUSTIFICATION;
+  }
+  else if (optarg == "lrb")
+  {
+    return decision::DECISION_STRATEGY_LRB;
+  }
+  else if (optarg == "help")
+  {
     puts(s_decisionModeHelp.c_str());
```
[the remark this answers is not in the document]
Comment (anwu1219) [reply src/options/options_handler.cpp]: How should I update the help string?

**Comment by 4tXJ7f** [comment]
Also, please add the licensing information of the code from external sources in the source file itself and in the COPYING file.

File: src/prop/minisat/core/Solver.h (hunk 1 above)
Comment (4tXJ7f) [reply src/prop/minisat/core/Solver.h]: Note: if my comment is correct and the variables are not initialized, we probably need to reevaluate the strategies affected by it.

Comment (anwu1219) [reply src/prop/minisat/core/Solver.h]: Yeah, I think you are correct that they are not initialized. This is probably why LRB did not win much. I think I need to re-evaluate two strategies: lrb and lrb+stoponly. The other ones should be fine.

**Comment by mpreiner** [comment]
As discussed in the last meeting we'll close this due to inactivity.

**Pull request closed without merging**